src/extension/importer/core/src/main/java/org/geoserver/importer/DefaultSupplementalFileExtensionsProvider.java

```diff
@@ -0,0 +1,46 @@
+/* (c) 2020 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.importer;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import org.apache.commons.lang.StringUtils;
+
+/** Default implementation of a SupplementalFileExtensionsProvider */
+public class DefaultSupplementalFileExtensionsProvider
+        implements SupplementalFileExtensionsProvider {
+    private Set<String> acceptedInputExtensions;
+    private Set<String> supplementalExtensions;
+    private Set<String> upperCaseSupplementalExtensions;
+
+    public DefaultSupplementalFileExtensionsProvider(
+            Set<String> acceptedInputExtensions, Set<String> supplementalExtensions) {
+        this.acceptedInputExtensions = Collections.unmodifiableSet(acceptedInputExtensions);
+        this.supplementalExtensions = Collections.unmodifiableSet(supplementalExtensions);
+        HashSet<String> upperCase = new HashSet<>();
```

Comment:
Better practice, declare interface, create instance.
```suggestion
        Set<String> upperCase = new HashSet<>();
```

Reply:
Fixed